### PR TITLE
Webapp registration styling fix

### PR DESCRIPTION
**Pull request by nfelger**
# Short Description
- The vertical spacing around the registration button was wrong on small viewports (see https://steuerlotse.atlassian.net/browse/STL-1398 for images).
- The horizontal spacing between button and explanatory text was also different from the designs, so I changed that, too.

# Changes
- FormNavButtons CSS
- I also changed the `http-proxy-middleware` proxy target host from `localhost` to `127.0.0.1` because that fixes a problem for me.
 
# Feedback
- Could you please test if proxying from react dev server (`yarn start`) to the Flask app still works for you with the changed setting?

### CODE REVIEW

**Review by frederike-ramin — APPROVED** [review]
It works and looks good (I personally would have taken a row-gap of 4 because of the "Gesetz der Nähe" that Nadine always preaches (just saying this because I saw that the concrete spacing wasn't defined) but I'm sure it's fine.)
starting with `yarn start` also works for me.

File: webapp/client/src/setupProxy.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ function setupProxy(app) {
         "!/*.hot-update.json", // HMR requests
       ],
       {
-        target: "http://localhost:5000",
+        target: "http://127.0.0.1:5000",
```
Comment (JulianRoesner) [inline webapp/client/src/setupProxy.js:17]: For me it works with both. So, it might be some local configuration change on your side 🤷

**Comment by nfelger** [comment]
Interesting, I didn't know about that Gesetz! The spacing value comes from Nadine herself, but maybe she thought I was asking about horizontal spacing instead of vertical. I asked her for confirmation on the ticket and will keep this unmerged until I've heard from her.

File: webapp/client/src/stories/FormNavButtons.stories.js [hunk 2]
```diff
@@ -21,6 +21,11 @@ WithOverviewLink.args = {
 export const WithExplanatoryText = Template.bind({});
 WithExplanatoryText.args = {
   ...Default.args,
-  explanatoryButtonText:
-    'Sie haben Ihren Freischaltcode bereits erhalten? <br><a href="#">Dann können Sie sich anmelden</a>',
+  explanatoryButtonText: (
+    <>
+      Sie haben Ihren Freischaltcode bereits erhalten?
+      <br />
```
Comment (JulianRoesner) [inline webapp/client/src/stories/FormNavButtons.stories.js:27]: I was curious why you had to change `<br>` to `<br />` and that was the result of my five minute research into the topic. 

- In HTML5 self-closing tags which can not have a child can use an optional slash at the end for compatibility with XHTML. However, the browser will ignore it. ([s. here](https://www.tutorialstonight.com/self-closing-tags-in-html.php))
- In Reacts/JSX you need to include the slash ([s. here](https://www.codecademy.com/courses/react-101/lessons/react-jsx-advanced/exercises/self-closing-tags)). Unfortunately, I could not find the reason besides that JSX is closer to XML than HTML. I am not going to invest more time into that but if you know it, let me know :) 


**TLDR: If you use self-closing tags like `<br>` in JSX, include the slash.**

**Review by JulianRoesner — APPROVED** [review]
👍 Looks good to me. `Yarn start` still works fine and the button looks good, too.

File: webapp/client/src/stories/FormNavButtons.stories.js (hunk 2 above)
Comment (nfelger) [reply webapp/client/src/stories/FormNavButtons.stories.js:27]: Good to know! This was an automatic change made by prettier.
